lightwood/mixer/sktime.py

```diff
@@ -95,32 +125,38 @@ def __call__(self, ds: Union[EncodedDs, ConcatedEncodedDs],
                            columns=['prediction'],
                            dtype=object)
 
-        data = {'data': ds.data_frame[self.target].reset_index(drop=True),
+        data = {'data': ds.data_frame[self.target],
                 'group_info': {gcol: ds.data_frame[gcol].tolist()
                                for gcol in self.grouped_by} if self.ts_analysis['tss'].group_by else {}}
 
-        # all_idxs = list(range(length))  # @TODO: substract, and assign empty predictions to remainder
-
-        for group in self.ts_analysis['group_combinations']:
-
-            if self.grouped_by == ['__default']:
-                series_idxs = data['data'].index
-                series_data = data['data'].values
-            else:
-                series_idxs, series_data = get_group_matches(data, group)
+        pending_idxs = set(range(length))
+        all_group_combinations = list(product(*[set(x) for x in data['group_info'].values()]))
+        for group in all_group_combinations:
+            series_idxs, series_data = get_group_matches(data, group)
 
             if series_data.size > 0:
+                group = frozenset(group)
+                series_idxs = sorted(series_idxs)
                 forecaster = self.models[group] if self.models[group].is_fitted else self.models['__default']
-
                 series = pd.Series(series_data.squeeze(), index=series_idxs)
-                series = series.sort_index(ascending=True)
-                series = series.reset_index(drop=True)
-
-                for idx, _ in enumerate(series.iteritems()):
-                    ydf['prediction'].iloc[series_idxs[idx]] = forecaster.predict(
-                        np.arange(idx, idx + self.n_ts_predictions)).tolist()
+                ydf = self._call_groupmodel(ydf, forecaster, series, offset=args.forecast_offset)
+                pending_idxs -= set(series_idxs)
 
-            if self.grouped_by == ['__default']:
-                break
+        # apply default model in all remaining novel-group rows
+        if len(pending_idxs) > 0:
+            series = pd.Series(data['data'][list(pending_idxs)].squeeze(), index=sorted(list(pending_idxs)))
+            ydf = self._call_groupmodel(ydf, self.models['__default'], series, offset=args.forecast_offset)
 
         return ydf[['prediction']]
+
+    def _call_groupmodel(self, ydf, model, series, offset=0):
```

Comment:
can you docstring this for the future maintainer ; )

Reply:
it's a private method, but this signature doesn't have type hints so it will be difficult to trace what needs to go where